client/src/main/java/org/apache/celeborn/client/write/DataPusher.java

```diff
@@ -213,7 +213,9 @@ protected void pushData(PushTask task) throws IOException {
 
   private void waitIdleQueueFullWithLock() throws InterruptedException {
     try {
-      while (idleQueue.remainingCapacity() > 0 && exceptionRef.get() == null) {
+      while (idleQueue.remainingCapacity() > 0
+          && exceptionRef.get() == null
+          && (pushThread != null && pushThread.isAlive())) {
```

Comment:
`pushThread` is responsible for managing `idleQueue`. If `pushThread` exits, theoretically the capacity of `idleQueue` will not change.

Reply:
> `pushThread` is responsible for managing `idleQueue`. If `pushThread` exits, theoretically the capacity of `idleQueue` will not change.

LGTM